boards/phytec/phycore_rt1170/phycore_rt1170.dtsi

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2024, PHYTEC America, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "phycore_rt1170-pinctrl.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	aliases {
+		led0 = &green_led;
+		led1 = &red_led;
+		sw0 = &user_button;
+		eeprom-0 = &eeprom0;
+		sdram0 = &sdram0;
+	};
+
+	sdram0: memory@80000000 {
+		/* MT48LC16M16A2B4-7EIT */
+		device_type = "memory";
+		reg = <0x80000000 DT_SIZE_M(64)>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		green_led: led-1 {
+			gpios = <&gpio13 12 GPIO_ACTIVE_HIGH>;
+			label = "Green LED D8";
+		};
+
+		red_led: led-2 {
+			gpios = <&gpio13 11 GPIO_ACTIVE_HIGH>;
+			label = "Red LED D7";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button: button-1 {
+			label = "User Button";
+			gpios = <&gpio10 2 GPIO_ACTIVE_HIGH>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
+	};
+};
+
+&lpuart1 {
+	pinctrl-0 = <&pinmux_lpuart1>;
+	pinctrl-1 = <&pinmux_lpuart1_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&lpuart5 {
+	pinctrl-0 = <&pinmux_lpuart5>;
+	pinctrl-1 = <&pinmux_lpuart5_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&lpuart6 {
+	pinctrl-0 = <&pinmux_lpuart6>;
+	pinctrl-1 = <&pinmux_lpuart6_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&lpi2c1 {
+	pinctrl-0 =<&pinmux_lpi2c1>;
+	pinctrl-names = "default";
+};
+
+&lpi2c2 {
+	pinctrl-0 =<&pinmux_lpi2c2>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	eeprom0: eeprom@50 {
+		compatible = "st,m24xxx", "atmel,at24";
+		reg = <0x50>;
+		status = "okay";
+		size = <256>;
+		pagesize = <8>;
+		address-width = <16>;
+		timeout = <5>;
+	};
+};
+
+&lpspi1 {
+	pinctrl-0 = <&pinmux_lpspi1>;
+	pinctrl-names = "default";
+};
+
+&flexspi {
+	pinctrl-0 = <&pinmux_flexspi1>;
+	pinctrl-names = "default";
+};
+
+&flexspi {
+	status = "okay";
+	ahb-prefetch;
+	ahb-read-addr-opt;
+	rx-clock-source = <1>;
+	reg = <0x400cc000 0x4000>, <0x30000000 DT_SIZE_M(16)>;
+	mx25u12832f: mx25u12832f@0 {
+		compatible = "nxp,imx-flexspi-nor";
+		/* MX25U12832FM2I02 is 16MB, 128MBit flash part */
+		size = <DT_SIZE_M(16 * 8)>;
+		reg = <0>;
+		spi-max-frequency = <133000000>;
```

Comment:
This isn't correct *unless* you correctly program the `DC[1:0]` bits to support fast reads at 133MHz. Looking at your flash datasheet:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/34665051/a84491de-4949-4140-8189-c3902952f7a5)

DC[1:0] needs to be set to `0b11`

The iMX RT1170 bootloader supports sending configuration commands using the flash configuration block. I'd look at the `configCmdSeqs` field of the flash configuration block: 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/34665051/be20cd15-e662-4e54-9c05-c5ab2c4a8988)

You should be able to set up the configuration command to send the "Write Status Register" command at boot, which will allow you to set the `DC` bits via the `cfgCmdArgs` parameter.

If you want an example of this, I'd suggest looking at the RT1170 EVK FCB: https://github.com/zephyrproject-rtos/hal_nxp/blob/7da0fc961f861b011b7f68a5085d29ee2b401957/mcux/mcux-sdk/boards/evkmimxrt1170/xip/evkmimxrt1170_flexspi_nor_config.c#L48. This configuration command sends command `0xC0`, which sets read parameters for the IS25WP flash to enable 133 MHz read operations